File: 소수_찾기.py
def solution(n):
    prime = [1] * (n + 1)
    
    for i in range(2, n + 1):
        if prime[i]:
            for j in range(i * 2, n + 1, i):
                prime[j] = 0
                
    return prime[2:].count(1)

File: 소수_찾기.cpp
#include <iostream>
using namespace std;

char prime[1000001];

int solution(int n) {
	for (int i = 2; i <= n; i++)
		prime[i] = 1;

	for (int i = 2; i <= n; i++)
		if (prime[i])
			for (int j = i * 2; j <= n; j += i)
				prime[j] = 0;

	int cnt = 0;
	for (int i = 2; i <= n; i++)
		if (prime[i]) cnt++;

	return cnt;
}
